### PR TITLE
feat: Changes to node cyclotron package

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Had a few gotchas along the way that I thought would be good to get sorted already. Not trying to do too much here

## Changes

* Move the cyclotron types and wrappers into the repo itself with a build step
* Actually link the package properly via file:
* Annoyingly `file:` linking doesn't take care of the build steps for us but this works well enough for now until we later 
* Moved initialisation into the cdp consumer start functions

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

**Review by bretthoerner — CHANGES_REQUESTED** [review]
Looks good other than a build issue I've detailed.

File: rust/cyclotron-node/package.json [hunk 1]
```diff
@@ -2,20 +2,25 @@
...
+    "main": "dist/index.js",
+    "types": "dist/index.d.ts",
     "scripts": {
         "test": "cargo test",
-        "cargo-build": "cargo build --message-format=json > cargo.log",
-        "cross-build": "cross build --message-format=json > cross.log",
-        "postcargo-build": "neon dist < cargo.log",
-        "postcross-build": "neon dist -m /target < cross.log",
-        "debug": "npm run cargo-build --",
-        "build": "npm run cargo-build -- --release",
```
[the remark this answers is not in the document]
Comment (benjackwhite) [reply rust/cyclotron-node/package.json:13]: 🤔 Interesting. It seemed to work fine from local testing.
`npm run build` is just proxying to exactly that cargo command and it ends up with all the files in the right place (tested it multiple times as it without this change the _only_ way it worked was via the docker image). It seemed to only be the `neon` part that errored out. Happy to change around the commands if there is something actually not working but moving things into npm commands has the benefit of making the whole inter-package operation work way better and avoids some of the hacks around imports.

[the remark this answers is not in the document]
Comment (bretthoerner) [reply rust/cyclotron-node/package.json:13]: > Interesting. It seemed to work fine from local testing.

Have you deleted the `index.node` dir so you have a clean slate?

[the remark this answers is not in the document]
Comment (bretthoerner) [reply rust/cyclotron-node/package.json:13]: Latest changes work for me locally and if I build the production image I seem to see the right stuff!

```
# find plugin-server/node_modules/cyclotron-node/
plugin-server/node_modules/cyclotron-node/
plugin-server/node_modules/cyclotron-node/index.node
plugin-server/node_modules/cyclotron-node/package.json
plugin-server/node_modules/cyclotron-node/dist
plugin-server/node_modules/cyclotron-node/dist/index.js
plugin-server/node_modules/cyclotron-node/dist/index.d.ts
plugin-server/node_modules/cyclotron-node/dist/index.js.map
```

Of course, tests seem mad, but I assume you see that:

```
FAIL tests/http-server.test.ts
  ● Test suite failed to run

    Cannot find module 'cyclotron-node' from 'src/cdp/cdp-consumers.ts'
```

**Pull request merged**